src/Html/HtmlGenerator.fs

```diff
@@ -173,19 +173,16 @@ module internal HtmlGenerator =
 
         let htmlType = bindingContext.ProvidedTypeDefinition(asm, ns, typeName, Some typeof<HtmlDocument>, hideObjectMethods = true, nonNullable = true)
         
-        let containerTypes = Dictionary<string, ProvidedTypeDefinition>()
+        let containerTypes = System.Collections.Concurrent.ConcurrentDictionary<string, ProvidedTypeDefinition>()
```

Comment:
Should it be? Maybe that is one problem of memory consumption that there is similar dictionaries in cache when executed every time separately in nested data structures.